src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,338 @@
+								if vhostWithMappings.Type != "primary" {
+									@componentStructural.RoundButton("ph-trash", "red-800", "red-600", "openDeleteVhostModal('"+vhostWithMappings.Hostname.String()+"')", "Delete virtual host", "")
+								}
+							</div>
+						</div>
+						<div class="flex space-x-3 mt-4">
+							@componentStructural.TagWithIcon("ph-folder-open", vhostWithMappings.RootDirectory.String(), "speedia-300")
+							@componentStructural.TagWithIcon("ph-file-text", vhostWithMappings.Type.String(), "speedia-300")
+						</div>
+					</div>
+					<div class="w-4"></div>
+					<!-- Mappings Column -->
+					<div class="bg-os-800 p-4 flex-col rounded w-1/1">
+						for _, mapping := range vhostWithMappings.Mappings {
+							@componentMappings.MappingsInfos(
+								mapping, vhostWithMappings.RootDirectory,
+								"openDeleteMappingModal("+mapping.Id.String()+", '"+mapping.Path.String()+"')",
+							)
+						}
+					</div>
+				</div>
+			}
+		</div>
+	</table>
+}
+
+templ VirtualHostRegularForm() {
+	<form
+		hx-post="/api/v1/vhosts/"
+		hx-indicator="#loading-overlay"
+		hx-swap="none"
+	>
+		<div class="my-6">
+			@componentForm.SelectInput("type", "Type", "virtualHost.type", valueObject.AvailableVirtualHostsTypes, false, false)
+		</div>
+		<template x-if="virtualHost.type == 'top-level'">
+			<div class="mb-6">
+				@componentForm.InputField("text", "hostname", "Hostname", "virtualHost.hostname", false)
+			</div>
+		</template>
+		<template x-if="virtualHost.type == 'subdomain'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "hostname", "Subdomain", "virtualHost.hostname", false)
+				@componentForm.InputField("text", "parentHostname", "Parent Hostname", "virtualHost.parentHostname", false)
+			</div>
+		</template>
+		<template x-if="virtualHost.type == 'alias'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "hostname", "Alias", "virtualHost.hostname", false)
+				@componentForm.InputField("text", "parentHostname", "Parent Hostname", "virtualHost.parentHostname", false)
+			</div>
+		</template>
+		<template x-if="virtualHost.hostname == ''">
+			@componentForm.SubmitButton("Create", "ph-check-fat", "closeCreateVhostModal()", false, true)
+		</template>
+		<template x-if="virtualHost.hostname != ''">
+			@componentForm.SubmitButton("Create", "ph-check-fat", "closeCreateVhostModal()", false, false)
+		</template>
+	</form>
+}
+
+templ CreateVhostModal() {
+	@componentStructural.Modal(
+		"Create Virtual Host", "isCreateVhostModalOpen", "closeCreateVhostModal()",
+		"",
+	) {
+		@VirtualHostRegularForm()
+	}
+}
+
+func filterOnlyVhostsHostnames(vhostsWithMappings []dto.VirtualHostWithMappings) []string {
+	vhostsHostnames := []string{}
+	for _, vhostWithMapping := range vhostsWithMappings {
+		vhostsHostnames = append(vhostsHostnames, vhostWithMapping.Hostname.String())
+	}
+
+	return vhostsHostnames
+}
+
+func getOnlyServiceNames() []string {
```

Comment:
Lol, here you used a direct name. Why would you call one filter and other get if they are basically the same thing?

Reply:
I differentiated the names because while the `get` method retrieves information from an external source, the `filter` method searches within a parameter that I provide. In other words, one is for gathering data, and the other is for filtering it.